cassandra/cqltypes.py

```diff
@@ -450,6 +450,8 @@ def serialize(dec, protocol_version):
             try:
                 sign, digits, exponent = Decimal(dec).as_tuple()
             except Exception:
+                if Decimal(dec).is_nan():
+                    raise ValueError("NaN values are not accepted values because they can't be serialized.")
```

Comment:
A test for this change would be a good idea.  tests:unit:test_types.py would be a good place to look for inspiration.